modules/nlu-testing/src/tools/visualisation.ts

```diff
@@ -0,0 +1,149 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import { distance, similarity } from 'ml-distance'
+import { Matrix } from 'ml-matrix'
+import { PCA } from 'ml-pca'
+
+import { BotState } from '../backend/typings'
+
+const clustering = require('density-clustering')
+
+export async function computeEmbeddingSimilarity(state: BotState) {
+  const intentDatas = _.groupBy(state.trainDatas, 'intent')
+
+  const intentEmb = {}
+  for (const key in intentDatas) {
+    if (Object.prototype.hasOwnProperty.call(intentDatas, key)) {
+      const meanEmb = intentDatas[key]
+        .map(o => o.utt_emb)
+        .reduce((tot, cur) => {
+          return Matrix.add(tot, new Matrix([cur]))
+        }, Matrix.zeros(1, intentDatas[key][0].utt_emb.length))
+
+      intentEmb[key] = Matrix.div(meanEmb, intentDatas[key].length).to1DArray()
+    }
+  }
+
+  const indexAndIntent = Array.from(Object.keys(intentEmb).entries())
+  const simMat = { matrix: [], labels: [] }
+  for (const [index, key] of indexAndIntent) {
+    const row = []
+
+    for (const [indexTodo, keyTodo] of indexAndIntent) {
+      if (index === indexTodo) {
+        row.push(1)
+      } else {
+        row.push(_.round(similarity.cosine(intentEmb[key], intentEmb[keyTodo]), 2))
+      }
+    }
+
+    simMat.labels.push(key)
+    simMat.matrix.push(row)
+  }
+
+  const plotlyMatrixData = [
```

Comment:
can't you compute what has to do with plotly in the frontend where Plotly actually is? At least, import a typing for this datastructure.